File: moviper-common/src/main/java/com/mateuszkoslacz/moviper/iface/view/ContextHolder.java
package com.mateuszkoslacz.moviper.iface.view;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;

import com.hannesdorfmann.mosby.mvp.MvpView;

/**
 * Created by mateuszkoslacz on 15.12.2016.
 * <p>
 * It's used for providing Context to the routing. Used in pair with {@link MvpView} in Moviper
 * views.
 * It's used in ViperService.
 */


public interface ContextHolder {

    @NonNull
    Context getContext();

    Bundle getArgs();
}

File: moviper3/src/main/java/com/mateuszkoslacz/moviper3/iface/view/ContextHolder.kt
package com.mateuszkoslacz.moviper3.iface.view

import android.content.Context
import android.os.Bundle

import com.hannesdorfmann.mosby3.mvp.MvpView

/**
 * Created by mateuszkoslacz on 15.12.2016.
 *
 *
 * It's used for providing Context to the routing. Used in pair with [MvpView] in Moviper
 * views.
 * It's used in ViperService.
 */


interface ContextHolder {

    // TODO
    fun getContext(): Context?

    val args: Bundle?
}
